hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -178,88 +178,118 @@ def boolean(data):
-                # property that shrinking from 2 to 1 can cause the result to
-                # grow, but the shrinker always tries 0 and 1 first anyway, so
-                # this will usually be fine.
-                result = bool(i)
+            elif p >= 1:
+                self.rig(data, True)
+                result = True
             else:
-                # Originally everything in the region 0 <= i < falsey was false
-                # and everything above was true. We swapped one truthy element
-                # into this region, so the region becomes 0 <= i <= falsey
-                # except for i = 1. We know i > 1 here, so the test for truth
-                # becomes i > falsey.
-                result = i > falsey
-        break
-    data.stop_example()
-    return result
+                remainder = opts * p - truthy
+
+                if falsey + truthy == opts:
+                    if isinstance(p, Fraction):
+                        m = p.numerator
+                        n = p.denominator
+                    else:
+                        m, n = p.as_integer_ratio()
+                    assert n & (n - 1) == 0, n  # n is a power of 2
+                    assert n > m > 0
+                    truthy = m
+                    falsey = n - m
+                    partial = False
+                else:
+                    partial = True
+
+                i = data.draw_bits(self.n_bits)
+
+                # We always label the region that causes us to repeat the loop
+                # is opts - 1 so that shrinking this byte never causes us to
+                # need to draw more data.
+                if partial and i + 1 == opts:
+                    p = remainder
+                    continue
+                if i <= 1:
+                    # We special case so that zero is always false and 1 is
+                    # always true which makes shrinking easier because we can
+                    # always replace a truthy block with 1. This has the
+                    # slightly weird property that shrinking from 2 to 1 can
+                    # cause the result to grow, but the shrinker always tries 0
+                    # and 1 first anyway, so this will usually be fine.
+                    result = bool(i)
+                else:
+                    # Originally everything in the region 0 <= i < falsey was
+                    # false and everything above was true. We swapped one
+                    # truthy element into this region, so the region becomes
+                    # 0 <= i <= falsey except for i = 1. We know i > 1 here, so
+                    # the test for truth becomes i > falsey.
+                    result = i > falsey
+            break
+        data.stop_example()
+        return result
+
+
+def biased_coin(data, p):
+    """Return False with probability p (assuming a uniform generator),
+    shrinking towards False.
+
+    If force is set to True or False then this will return that value
```

Comment:
What is `force` in context of this function?

Reply:
A red herring, thanks. I went down one route for changing this and then changed my mind and went with the `Coin` class instead.